src/main/java/github/scarsz/discordsrv/objects/managers/link/JdbcAccountLinkManager.java

```diff
@@ -53,7 +53,8 @@
 @SuppressWarnings("SqlResolve")
 public class JdbcAccountLinkManager extends AbstractAccountLinkManager {
 
-    private final static Pattern JDBC_PATTERN = Pattern.compile("([a-z]+)://(.+):(.+)/([A-z0-9]+)"); // https://regex101.com/r/7PSgv6
+    // more accurate RegEx for easier use later on and more accurate matches:
+    private final static Pattern JDBC_PATTERN = Pattern.compile("^(?<DBProtocol>[a-zA-Z]+):(?<DBEngine>[a-zA-Z]+)://(?<DBHost>.+):(?<DBPort>[0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])/(?<DBName>[a-zA-Z0-9_]+)(?<DBParameters>?[a-zA-Z0-9]+=.[a-zA-Z0-9]+(?:&[a-zA-Z0-9]+=[a-zA-Z0-9]+)*)?$");
```

Comment:
not really sure why Vankka recommended that you remove the backslashes. You actually need an *extra* backslash for it to work.
```suggestion
    private final static Pattern JDBC_PATTERN = Pattern.compile("^(?<DBProtocol>[a-zA-Z]+):(?<DBEngine>[a-zA-Z]+):\\/\\/(?<DBHost>.+):(?<DBPort>[0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])\\/(?<DBName>[a-zA-Z0-9_]+)(?<DBParameters>\\?[a-zA-Z0-9]+=.[a-zA-Z0-9]+(?:&[a-zA-Z0-9]+=[a-zA-Z0-9]+)*)?$");
```

Reply:
![image](https://user-images.githubusercontent.com/20226336/107014956-edfcd780-67a4-11eb-883f-46be589b39db.png)
Not in Java, the one for the question mark actually needs to be there though
```suggestion
    private final static Pattern JDBC_PATTERN = Pattern.compile("^(?<DBProtocol>[a-zA-Z]+):(?<DBEngine>[a-zA-Z]+)://(?<DBHost>.+):(?<DBPort>[0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])/(?<DBName>[a-zA-Z0-9_]+)(?<DBParameters>\\?[a-zA-Z0-9]+=.[a-zA-Z0-9]+(?:&[a-zA-Z0-9]+=[a-zA-Z0-9]+)*)?$");
```

Reply:
> wouldn't this be more accurate ?
  
No, with a single backslash it won't compile, double is the right amount but that is redundant because / isn't used in Java's flavor of regex